packages/donor/src/screens/questionnaire/QuestionnaireScreenContainer.tsx

```diff
@@ -52,42 +48,16 @@ export function QuestionnaireScreenContainer(
 
   const onSuccess = async () => {
     setIsLoading(true);
-
-    if (debugMode) {
-      console.log(
-        "Asked to book one of the following appointments: ",
-        appointmentToBookStore.appointmentIds
-      );
-    }
-
-    const bookAppointmentResponse =
-      await FirebaseFunctions.donorBookAppointment(
-        appointmentToBookStore.appointmentIds
+    const response = await bookAppointment.tryBookAppoitment(
+      props.bookedAppointment?.id
+    );
+    setIsLoading(false);
+    if (response.status === FunctionsApi.BookAppointmentStatus.SUCCESS) {
+      bookAppointment.onSuccessfulBooking(() =>
+        props.setBookedAppointment(response.bookedAppointment)
       );
-
-    switch (bookAppointmentResponse.status) {
-      case FunctionsApi.BookAppointmentStatus.NO_AVAILABLE_APPOINTMENTS:
-        setError(bookAppointmentResponse.status);
-        setIsLoading(false);
-        break;
-
-      case FunctionsApi.BookAppointmentStatus.SUCCESS:
```

Comment:
Same here, for example.
Extracting this to a dedicated hook is a nice idea, but should be in a dedicated PR.

Reply:
Ran a sweep on the entire PR, 
I don't believe there is any refactoring left in this PR.